### ISSUE TITLE
page-has-heading-one ignores headings visible only to screen readers

### ISSUE THREAD

**Issue by sorin-davidoi** [dequelabs/axe-core#2056]
Expectation: `page-has-heading-one` correctly handles heading visible only to screen readers

Actual: `page-has-heading-one` ignores headings visible only to screen readers

Motivation: This is a regression introduced in `3.5.0`

axe-core version: `3.5.1`

Steps to reproduce:

Given (rendered at https://output.jsbin.com/zixewojela):

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>JS Bin</title>
  <style type="text/css">
    .sr-only {
      border: 0; 
      clip: rect(0 0 0 0); 
      clip-path: polygon(0px 0px, 0px 0px, 0px 0px);
      -webkit-clip-path: polygon(0px 0px, 0px 0px, 0px 0px);
      height: 1px; 
      margin: -1px;
      overflow: hidden;
      padding: 0;
      position: absolute;
      width: 1px;
      white-space: nowrap;
    }
  </style>
</head>
<body>
  <main>
    <h1 class="sr-only">H1</h1>
  </main>
</body>
</html>
```

`axe https://output.jsbin.com/zixewojela --rules page-has-heading-one` reports no violations with `axe-core@3.4.2`, but reports one violation with with `3.5.1`:

```
  Violation of "page-has-heading-one" with 1 occurrences!
    Ensure that the page, or at least one of its frames contains a level-one heading. Correct invalid elements at:
     - html
    For details, see: https://dequeuniversity.com/rules/axe/3.5/page-has-heading-one
```

**Comment by danhumaan** [comment]
We are also running into this error in our testing and would love some clarity.

**Comment by straker** [comment]
I believe this is caused by [this fix](https://github.com/dequelabs/axe-core/pull/2001) to try to exclude hidden h1s from the check. Seems we didn't take into account visible only to screen readers.

**Comment by padmavemulapati** [comment]
No `page-has-heading-one` voilation found on the given test page.
![image](https://user-images.githubusercontent.com/47409579/76512695-2869d180-647b-11ea-8a38-1b3df74982b8.png)
also tried with the pain test file with given source:
no voilations observed (even with `page-has-heading-one` 
![image](https://user-images.githubusercontent.com/47409579/76512900-7aaaf280-647b-11ea-9f5f-0c57a4501112.png)
Verified with latest axe-coconut(4.4.1.07119)

**Comment by straker** [comment]
No docs needed
